spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -234,28 +241,137 @@ private String jobDesc() {
+   */
+  private DeleteOrphanFiles.Result deleteFilesStreaming(
+      Iterator<String> orphanFiles, SetAccumulator<Pair<String, String>> conflicts) {
+    List<String> samplePaths =
+        Lists.newArrayListWithCapacity(MAX_ORPHAN_FILE_PATHS_TO_RETURN_WHEN_STREAMING);
+    long filesCount = 0;
 
     if (deleteFunc == null && table.io() instanceof SupportsBulkOperations) {
-      deleteFiles((SupportsBulkOperations) table.io(), orphanFiles);
+      SupportsBulkOperations bulkIO = (SupportsBulkOperations) table.io();
+      Iterator<List<String>> fileGroups = Iterators.partition(orphanFiles, DELETE_GROUP_SIZE);
+
+      while (fileGroups.hasNext()) {
+        List<String> fileGroup = fileGroups.next();
+
+        // Collect sample paths before deleting
+        for (String path : fileGroup) {
+          if (samplePaths.size() < MAX_ORPHAN_FILE_PATHS_TO_RETURN_WHEN_STREAMING) {
+            samplePaths.add(path);
+          }
+        }
+
+        try {
+          bulkIO.deleteFiles(fileGroup);
+          filesCount += fileGroup.size();
+          LOG.info("Deleted {} files using bulk deletes", fileGroup.size());
+        } catch (BulkDeletionFailureException e) {
+          int deletedInBatch = fileGroup.size() - e.numberFailedObjects();
+          filesCount += deletedInBatch;
+          LOG.warn(
+              "Deleted only {} of {} files using bulk deletes",
+              deletedInBatch,
+              fileGroup.size(),
+              e);
+        }
+      }
     } else {
+      List<String> filesList = Lists.newArrayList(orphanFiles);
+      filesCount = filesList.size();
+
+      // Collect sample paths
+      for (int i = 0;
+          i < Math.min(filesList.size(), MAX_ORPHAN_FILE_PATHS_TO_RETURN_WHEN_STREAMING);
+          i++) {
+        samplePaths.add(filesList.get(i));
+      }
 
+      Tasks.Builder<String> deleteTasks =
+          Tasks.foreach(filesList)
+              .noRetry()
+              .executeWith(deleteExecutorService)
+              .suppressFailureWhenFinished()
+              .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc));
+
+      if (deleteFunc == null) {
+        LOG.info(
+            "Table IO {} does not support bulk operations. Using non-bulk deletes.",
+            table.io().getClass().getName());
+        deleteTasks.run(table.io()::deleteFile);
+      } else {
+        LOG.info("Custom delete function provided. Using non-bulk deletes");
+        deleteTasks.run(deleteFunc::accept);
+      }
+    }
+
+    LOG.info("Deleted {} orphan files", filesCount);
+
+    // Validate no conflicts after processing all partitions
+    validateNoConflicts(conflicts);
+
+    // Always add summary row for consistency (same format regardless of result size)
+    String summary = String.format("[Total removed: %d files.]", filesCount);
+    samplePaths.add(summary);
+
+    return ImmutableDeleteOrphanFiles.Result.builder().orphanFileLocations(samplePaths).build();
```

Comment:
We keep all of the path in the result in memory?

Does this mean, that we only decreased the memory footprint of the process to half?